client/CMusicHandler.h

```diff
@@ -60,6 +60,7 @@ class CSoundHandler: public CAudioBase
 
 	std::map<AudioPath, int> ambientChannels;
 	std::map<int, int> channelVolumes;
+	std::map<int, bool> channelPlaying;

```

Comment:
This approach won't work.
SDL can (and will) reuse sound channel ID's.

It works in your case (since no other sounds play) but if somebody attempts to reuse this method he might run into issues.

Instead you should set callback and close window when callback arrives.
See `BattleInterface::playIntroSoundAndUnlockInterface` for example